contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1975,15 +1977,204 @@ check_fulltext_exist(const char *schema_name, const char *table_name)
 	relid = get_relname_relid((const char *) table_name, schemaOid);
 
 
-	// Check if table exists
+	/* Check if table exists */
 	if (!OidIsValid(relid))
 		ereport(ERROR,
 			(errcode(ERRCODE_UNDEFINED_TABLE),
 				errmsg("relation \"%s\" does not exist",
 					table_name)));
 	
 	ft_index_name = get_fulltext_index_name(relid, table_name);
-	return ft_index_name != NULL;
+	if(ft_index_name == NULL)
+		return false;
+
+	ft_column_name_list = get_fulltext_indexed_columns(relid, ft_index_name);
+	/* A safety check to handle the case when the index definition changes from the current definition */
+	if(ft_column_name_list == NULL)
+	{
+		pfree(ft_index_name);
+		return false;
+	}
+
+	/* Check if column exists */
+	for(int i = 0; i < column_name->length; i++)
+ 	{
+		if(!check_column_list(relid, (char *)(column_name->elements[i]).ptr_value))
```

Comment:
Store `(char *)(column_name->elements[i]).ptr_value` in a variable and re use it in check and error message